### PR TITLE
Modernize plugin codebase

### PR DESCRIPTION
**Pull request by anatol**
Replace deprecated API with its new alternatives

### CODE REVIEW

**Comment by anatol** [comment]
cc @linux-china who also interested in this plugin development

File: res/META-INF/plugin.xml [hunk 1]
```diff
@@ -71,7 +71,7 @@
 	<actions>
 		<action
 				class="org.ziglang.action.NewZigFile"
-				icon="ZigIcons.ZIG_FILE"
+				icon="/icons/zig_file.png"
```
Comment (anatol) [inline res/META-INF/plugin.xml]: I do not think this is the best solution.

But I was not able to make kotlin class work here. For some reason plugin loader is unable to finds `ZigIcons.ZIG_FILE` here. It feels like files generated by kotlin are in a different class loader than the java one.

cc @ice1000 @linux-china for a better solution

**Comment by anatol** [comment]
`Build` and `Run` operations now work in the IDEA

File: res/META-INF/plugin.xml (hunk 1 above)
Comment (ice1000) [reply res/META-INF/plugin.xml]: I think you should keep the original code.

File: res/META-INF/plugin.xml [hunk 2]
```diff
@@ -87,7 +87,7 @@
 		</action>
 		<group
 				id="Zig.Tools"
-				icon="ZigIcons.ZIG_BIG_ICON"
+				icon="/icons/zig_icon.png"
```
Comment (ice1000) [inline res/META-INF/plugin.xml]: Ditto

File: src/icons/ZigIcons.java [hunk 3]
```diff
@@ -1,21 +0,0 @@
...
-import org.jetbrains.annotations.NotNull;
-
-import javax.swing.*;
-
-/**
- * Icon holder class
- *
- * @author ice1000, HoshinoTended
- */
-public interface ZigIcons {
```
Comment (ice1000) [inline src/icons/ZigIcons.java:13]: I strongly discourage you to delete it

File: src/org/ziglang/ZigLanguage.java [hunk 4]
```diff
@@ -1,23 +0,0 @@
...
-import org.jetbrains.annotations.Contract;
-import org.jetbrains.annotations.NotNull;
-
-import static org.ziglang.Zig_constantsKt.ZIG_EXTENSION;
-import static org.ziglang.Zig_constantsKt.ZIG_NAME;
-
-/**
- * @author ice1000
- */
-public class ZigLanguage extends Language {
```
Comment (ice1000) [inline src/org/ziglang/ZigLanguage.java:13]: I strongly discourage you to replace this class with Kotlin

**Review by ice1000 — COMMENTED** [review]
I have learned a lesson on Kotlin before.

The more Kotlin you use, the slower the code would be & the less smart features would work for your code.

Be careful of Kotlin.

File: src/icons/ZigIcons.java (hunk 3 above)
Comment (ice1000) [reply src/icons/ZigIcons.java:13]: I think you should keep the Java version.

**Comment by ice1000** [comment]
Kotlin is a well-designed language, but the IDE support and the compiler are mind-blowingly lame. I've seen at least 2k internal compiler errors before. Never seen that much of ICEs on a compiler like Kotlin.

**Comment by anatol** [comment]
I feel that this issue is related to class loaders somehow. I doubt that it is some Kotlin compiler issue.

For now, I can drop the change from the PR. But eventually I think, all the source code needs to be consistent and written in a singe language (i.e. Kotlin).

File: res/META-INF/plugin.xml (hunk 1 above)
Comment (anatol) [reply res/META-INF/plugin.xml]: Okay, I dropped this commit for now. It would be great to clarify how to make kotlin files work with icons.

**Review by ice1000 — APPROVED** [review]

**Comment by ice1000** [comment]
Thank you for your contribution!

**Comment by ice1000** [comment]
> I doubt that it is some Kotlin compiler issue.

I think it's because Kotlin objects are compiled with constant instances, while the interface has static members. But Idk.

Thank you very much for your contribution! I have uploaded an update to JB plugin center.

**Pull request merged**